source/NVDAObjects/window/edit.py

```diff
@@ -277,31 +280,34 @@ def _getFormatFieldAndOffsets(self,offset,formatConfig,calculateOffsets=True):
 			    formatField["text-position"] = TextPosition.BASELINE
 		if formatConfig["reportColor"]:
 			if charFormat is None: charFormat=self._getCharFormat(offset)
-			if charFormat.dwEffects & CFE_AUTOCOLOR:
-				rgb = GetSysColor(SysColorIndex.WINDOW_TEXT)
-				# Translators: The text color as reported in Wordpad (Automatic) or NVDA log viewer.
-				formatField["color"] = _("{color} (default color)").format(
-					color=colors.RGB.fromCOLORREF(rgb).name,
-				)
-			else:
-				rgb = charFormat.crTextColor
-				formatField["color"] = colors.RGB.fromCOLORREF(rgb)
-			if charFormat.dwEffects & CFE_AUTOBACKCOLOR:
-				rgb = GetSysColor(SysColorIndex.WINDOW)
-				# Translators: The background color as reported in Wordpad (Automatic) or NVDA log viewer.
-				formatField["background-color"] = _("{color} (default color)").format(
-					color=colors.RGB.fromCOLORREF(rgb).name,
-				)
-			else:
-				rgb = charFormat.crBackColor
-				formatField["background-color"] = colors.RGB.fromCOLORREF(rgb)
+			self._getFormatFieldColor(charFormat, formatField)
 		if formatConfig["reportLineNumber"]:
 			formatField["line-number"]=self._getLineNumFromOffset(offset)+1
 		if formatConfig["reportLinks"]:
 			if charFormat is None: charFormat=self._getCharFormat(offset)
 			formatField["link"]=bool(charFormat.dwEffects&CFM_LINK)
 		return formatField,(startOffset,endOffset)
-
+	
+	def _getFormatFieldColor(self, charFormat, formatField):

```

Comment:
Could this function get type hints?
Feel free to leave `charFormat` if an existing type does not exist.

This function should also be named `set` rather than `get`

```suggestion
	def _setFormatFieldColor(self, charFormat, formatField):
```

Reply:
alternatively, return a color, and background-color to be used to set `formatField` in the caller function.

Reply:
Fixed in bc4887832cf903beee219772e506fc8cd9a0f5b1. I have opted for the first suggestion, i.e. just renamed the function using "set" instead of "get".